src/aaudio/AudioStreamAAudio.cpp

```diff
@@ -54,26 +54,21 @@ static aaudio_data_callback_result_t oboe_aaudio_data_callback_proc(
     }
 }
 
+// This runs in its own thread.
+// Call error callback from a static function in case the stream gets deleted.
 static void oboe_aaudio_error_thread_proc(AudioStreamAAudio *oboeStream,
-                                          AAudioStream *stream,
                                           Result error) {
-    if (oboeStream != nullptr) {
-        oboeStream->onErrorInThread(stream, error);
+    LOGD("%s() - entering >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>", __func__);
+    oboeStream->requestStop();
+    if (oboeStream->getCallback() != nullptr) {
+        oboeStream->getCallback()->onErrorBeforeClose(oboeStream, error);
     }
-}
-
-// 'C' wrapper for the error callback method
-static void oboe_aaudio_error_callback_proc(
-        AAudioStream *stream,
-        void *userData,
-        aaudio_result_t error) {
-
-    AudioStreamAAudio *oboeStream = reinterpret_cast<AudioStreamAAudio*>(userData);
-    if (oboeStream != NULL) {
-        // Handle error on a separate thread
-        std::thread t(oboe_aaudio_error_thread_proc, oboeStream, stream, static_cast<Result>(error));
-        t.detach();
+    oboeStream->close();
```

Comment:
We should add more documentation on what can and cannot be queried both before and after the stream is closed.

Reply:
For example not deleting stream in onErrorBeforeClose

Reply:
The docs for queries allowed after close should be another PR.
I warned users now not to delete in this method.

Reply:
Opened issue #536